src/test/java/org/broadinstitute/hellbender/utils/iterators/AllLocusIteratorUnitTest.java

```diff
@@ -0,0 +1,58 @@
+package org.broadinstitute.hellbender.utils.iterators;
+
+import org.broadinstitute.hellbender.engine.AlignmentContext;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.pileup.ReadPileup;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+
+import static org.mockito.Mockito.*;
+
+public class AllLocusIteratorUnitTest extends BaseTest {
+
+    @Test
+    public void testIteratorReturnsPileupsForAllLoci() {
+        final SimpleInterval providedInterval = new SimpleInterval("1", 1, 25);
+        final List<AlignmentContext> providedPileups = Arrays.asList(
+                new AlignmentContext(new SimpleInterval("1", 5, 5),
+                        when(mock(ReadPileup.class).size()).thenReturn(10).getMock()),
+                new AlignmentContext(new SimpleInterval("1", 15, 15),
+                        when(mock(ReadPileup.class).size()).thenReturn(10).getMock()),
+                new AlignmentContext(new SimpleInterval("1", 19, 19),
+                        when(mock(ReadPileup.class).size()).thenReturn(10).getMock()),
+                new AlignmentContext(new SimpleInterval("1", 22, 22),
+                        when(mock(ReadPileup.class).size()).thenReturn(10).getMock()));
+        final List<Integer> providedLoci = Arrays.asList(5, 15, 19, 22);
+
+        final Iterator<AlignmentContext> locusIterator = new AllLocusIterator(providedInterval, providedPileups.iterator());
+
+        final List<AlignmentContext> returnedPileups = new ArrayList<>();
+        locusIterator.forEachRemaining(pileup -> returnedPileups.add(pileup));
+
+        Assert.assertEquals(returnedPileups.size(), providedInterval.size(), "Wrong number of pileup objects returned");
+
```

Comment:
Add an assertion here that specifies that this is inclusive of the staring position for the provided interval

Reply:
Done